gen/models/submodels/variable.py

```diff
@@ -64,7 +64,7 @@ def __init__(self, name, description=None):
 
 class variable(object):
     """Class holding information related to an Ada variable"""
-    def __init__(self, name, type, description=None, value=None, default_value=None):
+    def __init__(self, name, type, description=None, value=None, default_value=None, not_null=False):
```

Comment:
I don't think we want to modify `variable.py` at all. This file represents an Ada variable, and all the things associated with it. `not null` is not a thing for variables. It is only allowed for parameters (ie. variables passed to subprograms). So I would expect that `self.not_null` only lives in `parameter.py` right below `self.mode`.

Reply:
I think it is possible though, try adding `Y : not null Ref_Element := X;` to one of the .adb files here then compiling: https://learn.adacore.com/courses/advanced-ada/parts/resource_management/access_types.html#null-not-null-access

Reply:
Wow. You are right. That is cool!